pkg/statistics/handle/autoanalyze/refresher/refresher.go

```diff
@@ -166,6 +168,43 @@ func (r *Refresher) rebuildTableAnalysisJobQueue() error {
 							currentTs,
 						)
 						pushJobFunc(job)
+						// Skip the rest of the loop.
+						continue
+					}
+
+					// Only analyze the partition that has not been locked.
+					partitionDefs := make([]model.PartitionDefinition, 0, len(pi.Definitions))
+					for _, def := range pi.Definitions {
+						if _, ok := lockedTables[def.ID]; !ok {
+							partitionDefs = append(partitionDefs, def)
+						}
+					}
+					partitionStats := getPartitionStats(r.statsHandle, tblInfo, partitionDefs)
+					// If the prune mode is static, we need to analyze every partition as a separate table.
+					if pruneMode == variable.Static {
```

Comment:
If the partition prune mode is static, will optimizer merge global stats?

Reply:
No. We will generate the plan based on the statistics of each partition.